internal/metric/wrapper.go

```diff
@@ -0,0 +1,208 @@
+// (e.g. 2xx, 5xx, etc.)
+func latencyForCodeClass(it *monitoring.TimeSeriesIterator, codeClass string) (float64, error) {
+	var latency float64
+	for {
+		series, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return 0, errors.Wrap(err, "error when iterating through time series")
+		}
+
+		// Because the interval and the series aligner are the same, only one
+		// point is returned per time series.
+		if series.Metric.Labels["response_code_class"] == codeClass {
+			latency = series.Points[0].Value.GetDoubleValue()
+		}
+	}
+
+	return latency, nil
+}
+
+// calculateErrorResponseRate calculates the percentage of 5xx error response.
+//
+// It obtains all the successful responses (2xx) and the error responses (5xx),
+// add them up to form a 'total'. Then, it divides the number of error responses
+// by the total.
+// It ignores any other type of responses (e.g. 4xx).
+func calculateErrorResponseRate(it *monitoring.TimeSeriesIterator) (float64, error) {
+	var errorResponseCount, successfulResponseCount int64
+	for {
+		series, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return 0, errors.Wrap(err, "error when iterating through time series")
+		}
+
+		// Because the interval and the series aligner are the same, only one
+		// point is returned per time series.
+		switch series.Metric.Labels["response_code_class"] {
+		case "2xx":
+			successfulResponseCount += series.Points[0].Value.GetInt64Value()
+			break
+		case "5xx":
+			errorResponseCount += series.Points[0].Value.GetInt64Value()
+			break
+		}
+	}
+
+	totalResponses := errorResponseCount + successfulResponseCount
+	if totalResponses == 0 {
+		return 0, errors.New("no requests in interval")
+	}
+
+	rate := float64(errorResponseCount) / float64(totalResponses)
+
+	return rate, nil
+}
+
+func alignerAndReducer(alignReduceType alignReduce) (aligner monitoringpb.Aggregation_Aligner, reducer monitoringpb.Aggregation_Reducer) {
+	switch alignReduceType {
+	case Align99Reduce99:
+		aligner = monitoringpb.Aggregation_ALIGN_PERCENTILE_99
+		reducer = monitoringpb.Aggregation_REDUCE_PERCENTILE_99
+		break
+	case Align95Reduce95:
+		aligner = monitoringpb.Aggregation_ALIGN_PERCENTILE_95
+		reducer = monitoringpb.Aggregation_REDUCE_PERCENTILE_95
+		break
+	case Align50Reduce50:
+		aligner = monitoringpb.Aggregation_ALIGN_PERCENTILE_50
+		reducer = monitoringpb.Aggregation_REDUCE_PERCENTILE_50
+		break
+	}
+
+	return aligner, reducer
+}
+
+// NewFilter initializes a filter for a query.
+func NewFilter(serviceName, revisionName string) Filter {
+	filter := Filter{}
+	filter = filter.Add("resource.labels.service_name", serviceName)
+	filter = filter.Add("resource.labels.revision_name", revisionName)
+
+	return filter
+}
+
+// Add adds a filter for the query.
+func (f Filter) Add(key string, value string) Filter {
+	if f.query != "" {
+		f.query += " AND "
+	}
+	f.query += fmt.Sprintf("%s=%q", key, value)
```

Comment:
for now we probably just do `%s AND %s`, but in the future, this could benefit from an internal variable storing the filters, and a `Query() string` method compiling the internal state into a query string.

Reply:
I agree. I didn't want to "overengineer" here since we just need simple queries for now.